versioned/storage/jdbc/src/main/java/org/projectnessie/versioned/storage/jdbc/JdbcBackend.java

```diff
@@ -169,18 +187,24 @@ public PersistFactory createFactory() {
   Connection borrowConnection() throws SQLException {
     Connection c = dataSource.getConnection();
     c.setAutoCommit(false);
+    if (first.compareAndSet(true, false)) {
+      catalog = c.getCatalog();
+      schema = c.getSchema();
```

Comment:
Do we actually need these fields? I can only see them being used in `configInfo()` but there's no guarantee that `borrowConnection()` is called before `configInfo()`... WDYT?

Reply:
Actually, `PersistProvider` calls both `setupSchema()` and `configInfo()`... I wonder if we could have the config info as a return value from `setupSchema()`... WDYT?

Reply:
You are correct, and yes that could be a nice little improvement. Mind if I do this in a follow-up PR?

Reply:
sure :+1: